mobilecoind/src/service.rs

```diff
@@ -4184,6 +4292,226 @@ mod test {
+        assert!(!utxos.is_empty());
+
+        // We will try to take half of the offered input.
+        let mut sci_for_tx = api::SciForTx::new();
+        sci_for_tx.set_sci(generate_swap_response.get_sci().clone());
+        sci_for_tx.set_partial_fill_value(offered_value / 4);
+
+        // Try to add the swap to a mixed transaction
+        let mut request = api::GenerateMixedTxRequest::new();
+        request.set_sender_monitor_id(counterparty_monitor_id.to_vec());
+        request.set_change_subaddress(0);
+        // Only token id 1 is needed to fulfill the other side of the sci
+        request.set_input_list(
+            utxos
+                .iter()
+                .filter(|utxo| utxo.token_id == 1)
+                .map(Into::into)
+                .collect(),
+        );
+        request.set_scis(vec![sci_for_tx].into());
+        let generate_mixed_tx_response = client.generate_mixed_tx(&request).unwrap();
+
+        assert_eq!(
+            generate_mixed_tx_response
+                .get_tx_proposal()
+                .get_scis()
+                .len(),
+            1
+        );
+
+        let tx = Tx::try_from(generate_mixed_tx_response.get_tx_proposal().get_tx()).unwrap();
+
+        assert_eq!(tx.prefix.outputs.len(), 4);
+
+        let mut found_counterparty_output = false;
+        let mut found_counterparty_change = false;
+        let mut found_originator_output = false;
+        let mut found_originator_change = false;
+        for output in tx.prefix.outputs.iter() {
+            if let Ok((amount, _scalar)) =
+                output.view_key_match(swap_counterparty.view_private_key())
+            {
+                if amount.token_id == Mob::ID {
+                    assert_eq!(amount.value, offered_value / 4 - Mob::MINIMUM_FEE);
+                    found_counterparty_output = true;
+                } else {
+                    assert_eq!(*amount.token_id, 1);
+                    // It's not very easy to figure out what the value of this change output should
+                    // be, but seems not critical
+                    found_counterparty_change = true;
+                }
+            } else if let Ok((amount, _scalar)) =
+                output.view_key_match(swap_originator.view_private_key())
+            {
+                if amount.token_id == Mob::ID {
+                    assert_eq!(amount.value, offered_value * 3 / 4);
+                    found_originator_change = true;
+                } else {
+                    assert_eq!(amount, Amount::new(31, TokenId::from(1)));
+                    found_originator_output = true;
+                }
+            }
+        }
+        assert!(found_counterparty_output);
+        assert!(found_counterparty_change);
+        assert!(found_originator_output);
+        assert!(found_originator_change);
+
+        // Changing the fee token id to 1 should work, and slightly adjust the output
+        // values.
+        request.set_fee_token_id(1);
```

Comment:
Checking my understanding here: does this result in zero fee since there's no fee configured in the default fee map?
If so, maybe we want to change the fee map to include a fee for some non-mob tokens? I think this can be done [here](https://github.com/mobilecoinfoundation/mobilecoin/blob/mobilecoind-mixed-transactions/mobilecoind/src/test_utils.rs#L135-L138)

Reply:
I think that's right -- will look into this

Reply:
When I tried to actually test it, it's selecting an unexpectedly large fee, with just the default map, still trying to figure out why